src/Symfony/Component/Clock/MockClock.php

```diff
@@ -80,19 +93,47 @@ public function modify(string $modifier): void
      */
     public function withTimeZone(\DateTimeZone|string $timezone): static
     {
+        $this->now ??= $this->tryFromTimeProvider();
+
         if (\PHP_VERSION_ID >= 80300 && \is_string($timezone)) {
-            $timezone = new \DateTimeZone($timezone);
+            $this->timezone = new \DateTimeZone($timezone);
         } elseif (\is_string($timezone)) {
             try {
-                $timezone = new \DateTimeZone($timezone);
+                $this->timezone = new \DateTimeZone($timezone);
             } catch (\Exception $e) {
                 throw new \DateInvalidTimeZoneException($e->getMessage(), $e->getCode(), $e);
             }
         }
 
         $clone = clone $this;
-        $clone->now = $clone->now->setTimezone($timezone);
+        $clone->timeProvider = $this->timeProvider;
+        $clone->timezone = $this->timezone;
+        $clone->now = $clone->now->setTimezone($this->timezone);
 
         return $clone;
     }
+
+    public function reset(): void
+    {
+        if (null === $this->timeProvider) {
+            throw new \LogicException('The MockClock can only be reset when it was created with a time provider closure.');
+        }
+
+        $this->now = null;
+    }
+
+    private function tryFromTimeProvider(): DatePoint
+    {
+        if (null === $this->timeProvider) {
+            throw new \LogicException('The MockClock was not created with a time provider closure.');
+        }
+
+        $now = ($this->timeProvider)($this->timezone ?? new \DateTimeZone('UTC'));
+
+        if (!$now instanceof DatePoint) {
```

Comment:
why not allowing it to return a DateTimeImmutable and converting it, as done for the constructor signature ?

Reply:
Given that the `DatePoint` constructor is clock-aware, forcing a `DatePoint` instance might make things weird. 👍🏻 for allowing `DateTimeImmutable` instances.

Reply:
I agree, that's updated